src/Microsoft.OData.Core/ODataBatchJsonWriter.cs

```diff
@@ -0,0 +1,387 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataBatchJsonWriter.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+namespace Microsoft.OData.Core
+{
+    #region Namespaces
+    using System;
+    using System.Collections.Generic;
+    using System.Diagnostics;
+    using System.Globalization;
+
+#if ODATALIB_ASYNC
+    using System.Threading.Tasks;
+#endif
+
+    using Microsoft.OData.Core.Json;
+    using Microsoft.OData.Core.JsonLight;
+    #endregion Namespaces
+
+    /// <summary>
+    /// Class for writing OData batch messages of JSON type.
+    /// </summary>
+    internal sealed class ODataBatchJsonWriter : ODataBatchWriter

```

Comment:
>ODataBatchJsonWriter [](start = 26, length = 20)

Should probably be named "ODataJsonLightBatchWriter", and put in the "JsonLight" folder.